### PR TITLE
refactoring * using command pattern * using switch

### PR DESCRIPTION
**Pull request by 93artur**

### CODE REVIEW

File: src/main/java/com/arturobank/clientservice/ArturoBank.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class ArturoBank {
+
+    private Client myClient;
+    AccountBase accountBase = new AccountBase();
+    ClientBase clientBase = new ClientBase();
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/clientservice/ArturoBank.java:16]: do not forget about access modifiers pls!

File: src/main/java/com/arturobank/clientservice/ArturoBank.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private ClientOperation clientOperation = new ClientOperation();
+
+    public String getCurrentDateTime() {
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/clientservice/ArturoBank.java:19]: where do you use this method?

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+import com.arturobank.clientservice.*;
+import com.arturobank.datebase.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class ConsoleReader {
+
+    public boolean consistsOfLetters(String line) {
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/viewlayer/ConsoleReader.java:11]: "reader" - but has public validation method

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        if (line != null && line.matches("^[a-zA-Z]+$")) {
+            return true;
+        } else {
+            System.out.println("Invalid format");
+            return false;
+        }
+    }
+
+    public void showClientCommands() {
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/viewlayer/ConsoleReader.java:20]: "reader" by responsible for showing. So now this is ConsoleReaderAndShower

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    }
+
+    public void showStartCommands() {
+        System.out.println("1. Login");
+        System.out.println("2. Register");
+        System.out.println("3. Exit");
+        System.out.println("Enter your choice:");
+    }
+
+    public void registerClient(BufferedReader reader, ClientOperation clientOperation, ClientBase
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/viewlayer/ConsoleReader.java:33]: you do not use clientOperation

File: src/main/java/com/arturobank/clientservice/ArturoBank.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        ConsoleReader consoleReader = new ConsoleReader();
+        bank.clientOperation.registerClient("Katy", 1, "Katy", bank.accountBase.createAccountNumber(), bank.clientBase, bank.accountBase);
+        bank.clientOperation.registerClient("Egor", 2, "Egor", bank.accountBase.createAccountNumber(), bank.clientBase, bank.accountBase);
+        ExitCommand exitCommand = new ExitCommand();
+        ExitCommandOperation exitCommandOperation = new ExitCommandOperation(exitCommand);
+        CommandsExecutor commandsExecutor;
+        consoleReader.showStartCommands();
+        int commandNumber;
+        commandNumber = consoleReader.chooseStartCommand(reader);
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/clientservice/ArturoBank.java:38]: this is bad api! You don't have to send reader each time when you want to use consoleReader. Console reader have to have reader as a final field of the class

File: src/main/java/com/arturobank/datebase/AccountBase.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.arturobank.datebase;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AccountBase {
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/datebase/AccountBase.java:7]: this is really bad api!

File: src/main/java/com/arturobank/datebase/ClientBase.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package com.arturobank.datebase;
+
+import com.arturobank.clientservice.Client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ClientBase {
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/datebase/ClientBase.java:8]: this is bad api! This is not api at all!

**Review by IgorTsalko — CHANGES_REQUESTED** [review]

File: src/main/java/com/arturobank/clientservice/ArturoBank.java (hunk 1 above)
Comment (93artur) [reply src/main/java/com/arturobank/clientservice/ArturoBank.java:16]: fixed

File: src/main/java/com/arturobank/clientservice/ArturoBank.java (hunk 2 above)
Comment (93artur) [reply src/main/java/com/arturobank/clientservice/ArturoBank.java:19]: deleted it

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java (hunk 5 above)
Comment (93artur) [reply src/main/java/com/arturobank/viewlayer/ConsoleReader.java:33]: fixed

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java (hunk 4 above)
Comment (93artur) [reply src/main/java/com/arturobank/viewlayer/ConsoleReader.java:20]: fixed

File: src/main/java/com/arturobank/viewlayer/ConsoleReader.java (hunk 3 above)
Comment (93artur) [reply src/main/java/com/arturobank/viewlayer/ConsoleReader.java:11]: fixed

File: src/main/java/com/arturobank/datebase/AccountBase.java (hunk 7 above)
Comment (93artur) [reply src/main/java/com/arturobank/datebase/AccountBase.java:7]: fixed

File: src/main/java/com/arturobank/datebase/ClientBase.java (hunk 8 above)
Comment (93artur) [reply src/main/java/com/arturobank/datebase/ClientBase.java:8]: resolved

File: src/main/java/com/arturobank/viewlayer/ConsoleController.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.arturobank.viewlayer;
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/viewlayer/ConsoleController.java:1]: just view, not viewlayer

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
+package com.arturobank.datebase;
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/datebase/ClientBaseImpl.java:1]: this is "repository" layer

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    @Override
+    public Client getClientByAccountNumber(long accountNumber) {
+        Client client = null;
+        for (Long key : clientsOfBank.keySet()) {
+            if (key == accountNumber) {
+                client = clientsOfBank.get(key);
+                break;
+            }
+        }
+        return client;
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/datebase/ClientBaseImpl.java:30]: why do you need such hard logic? You have map with key as a accountNumber - then just get by key from the map
return clientsOfBank.get(key);

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    @Override
+    public Client getClientByUserName(String userName) {
+        Client client = null;
+        for (Long key : clientsOfBank.keySet()) {
+            if (clientsOfBank.get(key).getUserName().equals(userName)){
+                client = clientsOfBank.get(key);
+                break;
+            }
+        }
+        return client;
```
Comment (IgorTsalko) [inline src/main/java/com/arturobank/datebase/ClientBaseImpl.java:42]: the same

**Review by IgorTsalko — CHANGES_REQUESTED** [review]

File: src/main/java/com/arturobank/viewlayer/ConsoleController.java (hunk 9 above)
Comment (93artur) [reply src/main/java/com/arturobank/viewlayer/ConsoleController.java:1]: fied

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java (hunk 10 above)
Comment (93artur) [reply src/main/java/com/arturobank/datebase/ClientBaseImpl.java:1]: fixed

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java (hunk 11 above)
Comment (93artur) [reply src/main/java/com/arturobank/datebase/ClientBaseImpl.java:30]: resolved

File: src/main/java/com/arturobank/datebase/ClientBaseImpl.java (hunk 12 above)
Comment (93artur) [reply src/main/java/com/arturobank/datebase/ClientBaseImpl.java]: fixed

**Pull request merged**